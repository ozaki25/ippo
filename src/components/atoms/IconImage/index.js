import React from 'react';
import styled from 'styled-components';

const StyledIconImage = styled.img`
  border-radius: 50%;
  float: left;
  height: 48px;
  position: absolute;
  width: 48px;
`;

const IconImage = ({ ...props }) => <StyledIconImage {...props} />;

IconImage.displayName = 'IconImage';

export default IconImage;
