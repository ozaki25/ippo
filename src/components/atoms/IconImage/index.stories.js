import React from 'react';
import { storiesOf } from '@storybook/react';
import IconImage from '.';

const stories = storiesOf('atoms/IconImage', module);

const props = {
  src: '/icon.png',
};

stories.add('通常パターン', () => <IconImage {...props} />);
