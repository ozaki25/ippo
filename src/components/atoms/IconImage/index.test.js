import React from 'react';
import { snapshot } from 'src/test/helpers';
import IconImage from '.';

const props = () => ({
  src: '/icon.png',
});

snapshot('IconImage/nomal', <IconImage {...props} />);
