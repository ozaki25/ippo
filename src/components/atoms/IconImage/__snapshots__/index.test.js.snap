// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: IconImage/nomal 1`] = `
.c0 {
  border-radius: 50%;
  height: 48px;
  width: 48px;
}

<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <IconImage>
    <styled.img>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "lastClassName": "c0",
              "rules": Array [
                "
  border-radius: 50%;
  height: 48px;
  width: 48px;
",
              ],
            },
            "displayName": "styled.img",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "img",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <img
          className="c0"
        />
      </StyledComponent>
    </styled.img>
  </IconImage>
</MockRouter>
`;
